src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -22,56 +21,83 @@
     public static final String COMMAND_WORD = "deletefriend";
 
     public static final String MESSAGE_USAGE = COMMAND_WORD
-            + ": Deletes the friend identified by name.\n"
+            + ": Deletes the friend identified by index or name.\n"
             + "Parameters: "
-            + PREFIX_NAME + "NAME (must be name of a friend who exists in Amigos)\n"
-            + "Example: " + COMMAND_WORD + " " + PREFIX_NAME + "John Doe";
+            + PREFIX_NAME + "NAME (name of a friend who exists in Amigos) OR "
+            + "INDEX (a positive integer)\n"
+            + "Example 1: " + COMMAND_WORD + " " + PREFIX_NAME + "John Doe\n"
+            + "Example 2: " + COMMAND_WORD + " 1";
 
     public static final String MESSAGE_DELETE_PERSON_SUCCESS = "Deleted friend: %1$s";
 
-    //these dummy fields are needed to instantiate a person to be used to check if such a name entered exists
-    // in the address book
-    private static final Phone dummyPhone = new Phone("12345678");
-    private static final Email dummyEmail = new Email("dummyemail@gmail.com");
-    private static final Address dummyAddress = new Address("Dummy Address");
-
     private final Name nameOfPersonToDelete;
+    private final Index targetIndex;
+    private final boolean isDeletionByIndex;
 
-
+    /**
+     * Constructs a DeleteCommand for deletion by name
+     * @param nameOfPersonToDelete The name of the person to be deleted
+     */
     public DeleteCommand(Name nameOfPersonToDelete) {
         this.nameOfPersonToDelete = nameOfPersonToDelete;
+        this.targetIndex = Index.fromOneBased(1); //a dummy value
```

Comment:
Maybe can consider not using a dummy value? Could be unsafe due to ambiguity, unless you do regular sanity checks with `assert`.